core/src/main/scala/org/apache/spark/sql/execution/CachedPlanHelperExec.scala

```diff
@@ -86,7 +88,17 @@ case class CachedPlanHelperExec(childPlan: CodegenSupport, @transient session: S
           }
           alreadyGotBroadcastNode = true
         }
+        bchj transformAllExpressions {
+          case pl: ParamLiteral =>
+            session.getContextObject[mutable.Map[BroadcastHashJoinExec, ArrayBuffer[Any]]](
```

Comment:
No here it is like, if bchj is there but if it is not having ParamLiterals or constants then it is fine.